specs/merge/beacon-chain.md

```diff
@@ -239,6 +247,51 @@ def process_block(state: BeaconState, block: BeaconBlock) -> None:
 
 ### Execution payload processing
 
+#### `is_valid_gas_limit`
+
+```python
+def is_valid_gas_limit(payload: ExecutionPayload, parent: ExecutionPayloadHeader) -> bool:
+    parent_gas_limit = parent.gas_limit
+
+    # check if the payload used too much gas
+    if payload.gas_used > payload.gas_limit:
+        return False
+
+    # check if the payload changed the gas limit too much
+    if payload.gas_limit >= parent_gas_limit + parent_gas_limit // GAS_LIMIT_DIVISOR:
+        return False
+    if payload.gas_limit <= parent_gas_limit - parent_gas_limit // GAS_LIMIT_DIVISOR:
+        return False
+
+    # check if the gas limit is at least the minimum gas limit
+    if payload.gas_limit < MIN_GAS_LIMIT:
+        return False
+
+    return True
+```
+
+#### `compute_base_fee_per_gas`
+
+```python
+def compute_base_fee_per_gas(payload: ExecutionPayload, parent: ExecutionPayloadHeader) -> uint64:
+    parent_gas_target = parent.gas_limit // ELASTICITY_MULTIPLIER
+    parent_base_fee_per_gas = parent.base_fee_per_gas
+    parent_gas_used = payload.gas_used
+
+    if parent_gas_used == parent_gas_target:
+        return parent_base_fee_per_gas
+    elif parent_gas_used > parent_gas_target:
+        gas_used_delta = parent_gas_used - parent_gas_target
+        base_fee_per_gas_delta = \
+            max(parent_base_fee_per_gas * gas_used_delta // parent_gas_target // BASE_FEE_MAX_CHANGE_DENOMINATOR, 1)
+        return parent_base_fee_per_gas + base_fee_per_gas_delta
+    else:
+        gas_used_delta = parent_gas_target - parent_gas_used
+        base_fee_per_gas_delta = \
+            parent_base_fee_per_gas * gas_used_delta // parent_gas_target // BASE_FEE_MAX_CHANGE_DENOMINATOR
+        return max(parent_base_fee_per_gas - base_fee_per_gas_delta, 0)
```

Comment:
Good question. I'd remove `max` with a corresponding comment that it never underflows. It would make sense to leave this sanity check would this arithmetics be a subject to change to avoid potential underflow that could appear in the future. Even if `BASE_FEE_MAX_CHANGE_DENOMINATOR = 1` (which should never be the case) we got `0` in the "worst" case and never a `< 0` value.

Reply:
The check is removed and the comment is added

Reply:
I just realized that the unnecessary `max` was removed 2 days ago: https://github.com/ethereum/EIPs/pull/3640 😂

Reply:
lol 😂